cloud/aws.go

```diff
@@ -89,3 +101,58 @@ func GetAWSInstanceOnDemandHourlyPrice(instance *AWSInstance) (float64, bool, er
 
 	return pricePerHour, priceFound, nil
 }
+
+// GetAWSInstanceSpotHourlyPrice fetches current AWS spot price of an instance.
+func GetAWSInstanceSpotHourlyPrice(instance *AWSInstance) (float64, bool, error) {
+	var pricePerHour float64
+	var priceFound bool
+
+	sess := session.Must(session.NewSession(&aws.Config{Region: aws.String(instance.Region)}))
+	svc := ec2.New(sess)
+	input := &ec2.DescribeSpotPriceHistoryInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: aws.String("availability-zone"),
+				Values: []*string{
+					aws.String(instance.AvailabilityZone),
+				},
+			},
+			{
+				Name: aws.String("instance-type"),
+				Values: []*string{
+					aws.String(instance.Type),
+				},
+			},
+			{
+				Name: aws.String("product-description"),
+				Values: []*string{
+					aws.String("Linux/UNIX (Amazon VPC)"),
+				},
+			},
+		},
+	}
+
+	result, err := svc.DescribeSpotPriceHistory(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				fmt.Println(aerr.Error())
+			}
+		} else {
+			fmt.Println(err.Error())
+		}
+		return pricePerHour, priceFound, err
+	}
+
+	if len(result.SpotPriceHistory) > 0 {
+		pricePerHour, err = strconv.ParseFloat(*result.SpotPriceHistory[0].SpotPrice, 64)
```

Comment:
Are we using `*result.SpotPriceHistory[0]` because we only expect there to be one result set?

Reply:
Yes, based on the filters, yes. Although we'll need to fetch the platform type to pass it in (currently is static).

Reply:
> Although we'll need to fetch the platform type to pass it in (currently is static).

Specifically, this line ... https://github.com/Intellection/chargeback/pull/10/files#diff-b0956013f03620dc8cc50d9ade0cd501R150.